tests/e2e/suite_test.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/dynamic"
+
+	numaflowv1 "github.com/numaproj/numaflow/pkg/apis/numaflow/v1alpha1"
+	apiv1 "github.com/numaproj/numaplane/pkg/apis/numaplane/v1alpha1"
+	planeversiond "github.com/numaproj/numaplane/pkg/client/clientset/versioned"
+	planepkg "github.com/numaproj/numaplane/pkg/client/clientset/versioned/typed/numaplane/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+func TestE2E(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	RunSpecs(t, "E2E Suite")
+}
+
+var (
+	dynamicClient dynamic.DynamicClient
```

Comment:
Discussed with @juliev0 today, we should be able to use the Kubernetes client or Numaflow clients directly instead.

Reply:
Yeah, I was thinking using the kubernetes client seems to be a typical thing to do for kubernetes objects, as it's done in Numaflow. 

As for Numaflow, we could either use the dynamic client or the numaflow-specific client. Since we control the specs in our test, we would be okay to use the numaflow-specific client right now if we want.

**However**, once the test starts needing to run Pipelines of different Numaflow Controller versions and their varying Pipeline specs, we may want to use the dynamic client for that anyway. So, perhaps we could just continue to use dynamic client for Numaflow resources?

Reply:
That would make more sense to keep the dynamic client for specifically Pipelines/ISBServices/Vertices then. 
As for Kubernetes objects, we only look at `Deployment` to check that the NumaflowController is deployed and created. I could change this to use the Kubeclient if that would be best.

Reply:
Sounds good to me.